artifacts/mediaTypes.md

```diff
@@ -0,0 +1,211 @@
+
+## Config Media Types
+
+By supporting new `manifest.config.mediaTypes`, registry tools MAY understand the types stored. 
+
+A simplified example of a registry listing may look like:
+
+| artifact reference | icon | type |
+|-|-|-|
+| `samples/image/hello-world:1.0` |![](../img/oci-container.png)| container image | 
+| `samples/image/hello-world-doc:1.0`|![](../img/doc.png)| doc |
+| `samples/helm/hello-world:1.0` |![](../img/helm.png)| helm chart | 
+
+## Authoring New Artifact Types
+
+Artifact authors have the freedom to define their artifacts as meets their needs. To conform to the OCI Image spec, and be stored within an OCI distribution based registry, Artifact authors MUST follow the following requirements:
+
+- Artifact authors MUST provide unique `config.mediaTypes` to represent their artifact. 
+- Artifact authors MAY submit a PR for their `config.mediaType` to [./mediaTypeMappings.json]() to make their new artifact type well known.
+- Artifact authors MAY provide new schemas for the config.json object. 
+- Artifacts and tooling that use existing `config.mediaTypes` MUST conform to the artifacts spec.
+- Tools MAY choose to support new `config.mediaTypes`.
+- Tools MUST be able to ignore `config.mediaTypes` they don't support. 
+- Distribution implementations MAY choose to support new `config.mediaTypes`. See the [distribution-spec] for more details on processing of the `config.mediaTypes`
+
+### mediaTypes formatting
+
+New mediaTypes MUST use the following convention:
+
+`application/vnd.`[governingOrg/vendor]`.`[artifactName]`.config.`[version]`+json` 
+
+**Example:** 
+The Helm 3 artifact may be represented as: `application/vnd.cncf.helm/config.v3+json`
+
+## Config Parsing
+
+New artifacts MUST choose from 3 options for how the `config` object is processed and parsed:
+
+- Support the [`application/vnd.oci.image.config.v1+json`](../config.md) schema
+- Define a new schema, which artifact specific tools would need to support
+- Define a `null` schema, where the artifact only uses the `config.mediaType` to identify it's unique artifact type. 
+
+Artifact authors MAY publicize their schema in [./mediaTypeMappings.json]() through a pull request.
+
+Artifact authors understand supporting additional artifact types is optional, and distribution parsing of new config schemas is optional. However, new config schemas may be useful to that artifact tooling experience. 
+
+## Layer mediaTypes
+
+While OCI container images have ordinal layers, supporting overlaying of a unified file system, new artifact authors MAY define how they persist their layers, which may be individual files, or collections of files. The tooling specific to that artifact type owns the processing of the files and how they are extracted on the client. 
+
```

Comment:
Here is sounds like I have freedom to define custom types for the layers? And then it follows that with custom content types, the distribution spec doesn't have to server containers at all?

Reply:
The custom layer types are still layers. What I'm trying to say here is they don't have to culminate in a unified file system where all the layers overlay each other. They can be completely separate collections the artifact tool can extract as they wish. A new artifact may extract layer0 into the current directory, while it might extract layer1 into a per-user config directory. 
My next round is to do a PR on the distribution spec. What this should be conveying is distribution would accept and server layers. As long as it's a blob, like using the ORAS (https://github.com/deislabs/oras/) client, the registry shouldn't care. 
Does that clarify it?

Reply:
Yes I understand that bit - and maybe my question would be better asked on your next PR to the distribution spec.  What I am curious about is if the user is allowed custom mediaTypes (for example, docs) to be served in an image-spec (for download / otherwise handling by some client) then what is stopping someone from only serving (non container layer) artifacts in an image spec served by a registry? Technically, couldn't a registry exclusively serve one or more types, none of which are filesystem / layer related at all?

Reply:
Yes. The artifact registry enhancements don't "judge" what's being saved, as long as they follow some basic semantics to be reasoned over. For instance, if a team was doing some work and wanted to store foo artifacts in a registry, leveraging all the investments made in that PaaS or product; and either don't save any images, or they choose to save them in a different registry instance, have at it.

Reply:
That's really neat :) It makes the spec useful beyond what it was originally intended for.